System/RESTRouter.php

```diff
@@ -262,27 +263,34 @@ protected static function ErrorHandling(Exception $exception)
 
 			if ($exception instanceof SqlFriendlyException)
 			{
+				$debugMode = false;
 				$debugType = 'SQL';
+				$debugModeError = $exception->getMessage();
 			}
 			elseif (isset($config))
 			{
+				$debugMode = $config->DebugMode;
 				$debugModeError = $config->DebugModeError;
 				$debugType = $config::Alert;
 			}
 		}
 
 		if (static::$JSONErrors)
 		{
-			$error = array(
-				//'code' => $exception->getCode(),
-				'type' => $debugType ?: (($resourceException) ? $exception->GetErrorType() : get_class($exception)),
-				'message' => $debugModeError ?: $exception->getMessage()
-			);
+			$error = $debugMode ?
+				array('type' => get_class($exception), 'message' => $exception->getMessage()) :
+				array('type' => $debugType, 'message' => $debugModeError);
+
+			if ($resourceException)
+			{
+				$error['type'] = $exception->GetErrorType();
```

Comment:
I don't like this kind of pattern, it does a bunch of logic and then below just undoes all of it. A reader works top-down, so they are reading the top going "wow really?" trying to understand only to find out later none of that mattered. We should just have correct logic to begin with. Let's clean up and also put each key on its own line. Vertical space is free, reading/understanding is the guiding value.

Reply:
But I am happy you got rid of those gross unreadable conditions!!